src/components/Chat/Chat.ts

```diff
@@ -0,0 +1,21 @@
+import Block from "../../core/Block";
+import template from "./Chat.tmpl";
+import {withSelectedChat} from "../../hocs/withSelectedChat";
+
+export type ChatType = {
+    contactName: string,
+    selectedChat: any;
+    id: number;
+}
+
+export class ChatComponent extends Block<ChatType> {
+    public render(): DocumentFragment {
+        return this.compile(template, {
+            ...this.props,
+            isSelected: this.props.id === this.props.selectedChat?.id,
+        });
+    }
+}
+
+// @ts-ignore
+export const Chat = withSelectedChat(ChatComponent);
```

Comment:
Вместо отключение TS попробуйте использовать приведение типов 
```suggestion
export const Chat = withSelectedChat(ChatComponent as typeof Block);
```